gk/fib.c

```diff
@@ -1382,8 +1372,41 @@ check_prefix_locked(struct ip_prefix *prefix,
 	struct gk_fib *ip_prefix_fib;
 	struct gk_lpm *ltbl = &gk_conf->lpm_tbl;
 
-	if (action == GK_DROP || action == GK_FWD_GRANTOR)
+	if (action == GK_DROP || action == GK_FWD_GRANTOR) {
+		int num_rules;
+
+		if (prefix->addr.proto == ETHER_TYPE_IPv4) {
+			struct rte_lpm_rule_entry re_tbl[
+				gk_conf->max_num_ipv4_rules];
+			num_rules = rte_lpm_rule_iterator(
```

Comment:
I think the memory isn't an issue, considering each rule entry has at most 18 Bytes (IPv6), even if we use a large number of entries, e.g., 100K, then it will only consume (18 * 100K / 1M = 1.75MB). However, if iterating one entry per call, then we will have many calls (e.g., 100K), which will incur additional overhead.

Reply:
I think I'm more concerned with the cache performance when creating such a large array, and the fact that the hash library implements an iterator that retrieves one entry at a time makes me think this is the right way to do it for potentially long tables. But I don't have the knowledge to prove that theory, it's just a hunch, so I don't think I can push this issue any further.

I don't have any more suggestions -- thanks Qiaobin!

Reply:
I see. That's really a good point! It works in a more ``iterator`` way. Thanks Cody!